### ISSUE TITLE
Feature request: Option to run fd in different directory than pwd

### ISSUE THREAD

**Issue by hajdamak** [sharkdp/fd#475]
Git like -C option would be handy to run fd as if it was started in specified path rather than the current working directory.

**Comment by sharkdp** [comment]
Thank you for the feedback.

What is the use case for this?

Have you considered alternative options?

* `fd … some/other/directory`. This is not quite the same, as the paths will be prefixed with `some/other/directory`, but it's close. If you are interested in absolute paths (`-a`/`--absolute-path`), the result would be the same as with a `-C` option.
* Use a subshell `(cd some/other/directory; fd …)`

**Comment by sharkdp** [comment]
We're most probably not going to implement this in `fd`.

**Comment by hajdamak** [comment]
I am using fd output to feed [skim](https://github.com/lotabout/skim) on large directories. I need this to be as instant as possible. Having base directory in all entries makes fuzzy search a lot less efficient.

I am using this in [Fish](https://github.com/fish-shell/fish-shell) script which does not support subshell and using cd or pushd/popd messes up Fish history and makes back/forward keyboard shortcuts less efficient.

I have tried to pipe fd output to sed/sd before feeding skim in order to remove base directory from the list but this gets quite slow.

**Comment by sharkdp** [comment]
Ok, but a simple wrapper script would work, right?

Something like

`run-fd-in.sh`
``` bash
#!/bin/bash

set -e

base_directory="$1"
shift

cd "$base_directory"
fd "$@"
```

which can be used like
```
./run-fd-in.sh <base-directory> [fd-options…]
```

**Comment by hajdamak** [comment]
Yeah, that should work. I hoped to avoid spawning another shell though.
Just wondering, are there any other arguments against this functionality besides KISS, which I totally understand.

**Comment by sharkdp** [comment]
>  Just wondering, are there any other arguments against this functionality besides KISS, which I totally understand.

No, I'm just trying to keep `fd` simple (see beginning of README).

There is one compromise which I would be willing to make because this should actually be really simple to implement: we add a new "long" command-line option (not `-C`, but maybe `--base-directory`) as a hidden option that would not show up in the `-h` help text (and maybe not even in the `--help` text). If those are just a few lines of code along the lines of `if (base-directory option is present) { set_current_working_directory(…); }`, I'd be willing to include this in `fd`.

**Comment by hajdamak** [comment]
I have started implementing this and current implementation seems strange. Whether fd lists result with absolute or relative path does not depend on whether `path` parameter was passed or not. It depends whether `path` is set to `.` or something else. So for example if you execute `fd example . ~/work` in `inbox` directory, matches from `inbox` will be relative to `inbox` but matches from `work` will be absolute. Is this intentional?

**Comment by hajdamak** [comment]
Pull request #509 created.

**Comment by sharkdp** [comment]
closed via #509

**Comment by sharkdp** [comment]
Released in [fd 7.5.0](https://github.com/sharkdp/fd/releases/tag/v7.5.0)
